src/xtopcom/xrpc/xratelimit/xratelimit_cache.cpp

```diff
@@ -32,6 +32,7 @@ RatelimitCache::RatelimitCache(const RatelimitConfig* config)
 RatelimitCache::~RatelimitCache() {
     break_out_ = true;
     cv_.notify_all();
+    thread_.join();
```

Comment:
add an assertion or an 'if' condition check to make sure thread_ is joinable (for a non-joinable thread calling join() will throw an exception and this happens in the destructor) since join() API is public and can be called anywhere who holds the thread_ object. 

https://en.cppreference.com/w/cpp/thread/thread/join

Reply:
done